src/critic_manager.cpp

```diff
@@ -21,22 +22,23 @@ void CriticManager::on_configure(
 void CriticManager::getParams()
 {
   auto node = parent_.lock();
-  auto getParam = utils::getParamGetter(node, name_);
-  getParam(critic_names_, "critics", std::vector<std::string>{});
+  auto getParam = parameters_handler_->getParamGetter(name_);
+  getParam(critic_names_, "critics", std::vector<std::string>{}, ParameterType::Static);
 }
 
 void CriticManager::loadCritics()
 {
-  loader_ = std::make_unique<pluginlib::ClassLoader<critics::CriticFunction>>(
-    "mppic", "mppi::critics::CriticFunction");
+  if (!loader_) {
+    loader_ = std::make_unique<pluginlib::ClassLoader<critics::CriticFunction>>(
+      "mppic", "mppi::critics::CriticFunction");
+  }
 
-  critics_.clear();
```

Comment:
You need the clear so that if this is brought down by lifecycle and brought up again that the original set of critics are removed and then we only have the new ones being set. Else, we have duplicates setup

Reply:
forgot about this. returned. thanks